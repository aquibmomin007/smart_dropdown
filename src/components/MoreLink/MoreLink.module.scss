.dropdownMoreList{
    cursor: pointer;
    text-align: right;
    color: blue;
    font-size: 12px;
    margin: 5px 10px;

    &:hover{
        text-decoration: underline;
    }
}