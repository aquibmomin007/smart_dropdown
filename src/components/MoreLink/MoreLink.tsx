import React, {  } from 'react';
import styles from './MoreLink.module.scss';

export type MoreLinkProps = {
  moreValue: number;
  handleSetAllCountries: () => void;
}

const MoreLink = (props: MoreLinkProps) => { 
  const { moreValue, handleSetAllCountries } = props;
  return (
    <p 
        className={styles.dropdownMoreList} 
        onClick={handleSetAllCountries}
    >
        {`${moreValue} more...`}
    </p>
  )
}

export default MoreLink;