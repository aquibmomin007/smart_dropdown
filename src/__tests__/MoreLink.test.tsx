import * as React from 'react';
import MoreLink, {MoreLinkProps} from '../components/MoreLink/MoreLink';
import { render } from '@testing-library/react';


describe('MoreLink Testing', () => {
  test('renders correctly', () => {
    const moreLinkProps:MoreLinkProps = {
        moreValue: 5,
        handleSetAllCountries: jest.fn()
      };

    const { container, getByTestId } = render(
        <MoreLink {...moreLinkProps} />
    )
    expect(container.firstChild).toMatchSnapshot();
  });
});
