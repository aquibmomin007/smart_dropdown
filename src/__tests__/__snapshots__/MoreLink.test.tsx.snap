// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MoreLink Testing renders correctly 1`] = `
<p
  class="dropdownMoreList"
>
  5 more...
</p>
`;
